src/rgw/rgw_rest_s3.cc

```diff
@@ -2602,6 +2602,8 @@ int RGWSetRequestPayment_ObjStore_S3::get_params()
     return r;
   }
 
+  in_data.append(data);
```

Comment:
can SetRequestPayment use `data` directly, instead of adding a separate `in_data`?